src/main/java/plugins/nate/smp/listeners/PvPToggleListener.java

```diff
@@ -0,0 +1,87 @@
+package plugins.nate.smp.listeners;
+
+import org.bukkit.entity.Arrow;
+import org.bukkit.entity.Player;
+import org.bukkit.entity.ThrownPotion;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityDamageByEntityEvent;
+import org.bukkit.event.entity.PotionSplashEvent;
+import org.bukkit.event.player.PlayerQuitEvent;
+import plugins.nate.smp.managers.PvPManager;
+import plugins.nate.smp.tables.PlayerSettings;
+import plugins.nate.smp.utils.ChatUtils;
+import plugins.nate.smp.utils.PvPStatusUpdater;
+
+import static plugins.nate.smp.utils.ChatUtils.sendMessage;
+
+public class PvPToggleListener implements Listener {
+    @EventHandler
+    public void onPlayerOnPlayerAttack(EntityDamageByEntityEvent event) {
+        if (!(event.getEntity() instanceof Player victim)) {
+            return;
+        }
+
+        Player attacker = null;
+
+        if (event.getDamager() instanceof Player player) {
+            attacker = player;
+        } else if (event.getDamager() instanceof Arrow arrow && arrow.getShooter() instanceof Player player) {
+            attacker = player;
+        }
+
+        if (attacker == null || attacker.hasPermission("smp.bypasspvp")) {
+            return;
+        }
+
+        if (!PlayerSettings.getPvPStatus(attacker.getUniqueId())) {
+            event.setCancelled(true);
+            sendMessage(attacker, ChatUtils.PREFIX + "&cYou have pvp disabled!");
+            return;
+        }
+
+        if (!PlayerSettings.getPvPStatus(victim.getUniqueId())) {
+            event.setCancelled(true);
+            sendMessage(attacker, ChatUtils.PREFIX + "&c" + victim.getName() + " has pvp disabled!");
+            PvPManager.enterCombat(attacker);
+            PvPStatusUpdater.updatePlayerTitle(attacker);
+        } else {
+            PvPManager.enterCombat(attacker);
+            PvPManager.enterCombat(victim);
+            PvPStatusUpdater.updatePlayerTitle(attacker);
+            PvPStatusUpdater.updatePlayerTitle(victim);
+        }
+    }
+
+    @EventHandler
+    public void onPotionSplash(PotionSplashEvent event) {
+        ThrownPotion thrownPotion = event.getPotion();
+
+        boolean hasNegativeEffect = thrownPotion.getEffects().stream().anyMatch(effect -> PvPManager.negativeEffects.contains(effect.getType()));
+
+        if (!hasNegativeEffect || !(thrownPotion.getShooter() instanceof Player thrower)) {
+            return;
+        }
```

Comment:
Do the check for player thrower first, then calculate if the potion has a negative effect afterwards, instead of wasting work in the event this isn't a player.

```suggestion
        if (!(thrownPotion.getShooter() instanceof Player thrower)) {
            return;
        }
        
        boolean hasNegativeEffect = thrownPotion.getEffects().stream().anyMatch(effect -> PvPManager.negativeEffects.contains(effect.getType()));
        
        if (!hasNegativeEffect) {
            return;
        }
```